### PR TITLE
Make Unix platforms build with AWS-LC

### PR DESCRIPTION
**Pull request by andrew-kaufman**
*Issue #, if available:* #61 

*Description of changes:* This makes Unix platforms build only with AWS-LC and removes support for older versions of OpenSSL's libcrypto. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinboswell — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,8 +80,8 @@ else ()
         file(GLOB AWS_CAL_OS_SRC
             "source/unix/*.c"
         )
-        find_package(LibCrypto REQUIRED)
-        set(PLATFORM_LIBS LibCrypto::Crypto dl)
+        find_library(AWS_LC crypto REQUIRED)
+        set(PLATFORM_LIBS dl ${AWS_LC})
```
Comment (graebm) [inline CMakeLists.txt:84]: trivial: What's more normal for Cmake? To define a variable, or just have a normal fixed string?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -79,7 +81,7 @@ jobs:
         echo "${{ secrets.GITHUB_TOKEN  }}" | docker login docker.pkg.github.com -u awslabs --password-stdin
         export DOCKER_IMAGE=docker.pkg.github.com/awslabs/aws-crt-builder/aws-crt-${{ env.LINUX_BASE_IMAGE  }}:${{ env.BUILDER_VERSION  }}
         docker pull $DOCKER_IMAGE
-        docker run --env GITHUB_REF $DOCKER_IMAGE --version=${{ env.BUILDER_VERSION  }} build -p ${{ env.PACKAGE_NAME  }} --compiler=gcc-8 --spec=downstream cmake_args='-DBUILD_SHARED=ON'
+        docker run --env GITHUB_REF $DOCKER_IMAGE --version=${{ env.BUILDER_VERSION  }} build -p ${{ env.PACKAGE_NAME  }} --compiler=gcc-8 --spec=downstream 'cmake_args=-DBUILD_SHARED_LIBS=ON'
```
Comment (graebm) [inline .github/workflows/ci.yml:84]: I don't think the builder is handling cmake_args correctly yet.
When I "fixed" it the other day, I'd only tested in aws-c-common (which has no upstream dependencies).
When I tried it in aws-c-io, it didn't work.
I think you're hitting the same issue, where aws-c-common isn't getting built with -DBUILD_SHARED_LIBS=ON

File: CMakeLists.txt (hunk 1 above)
Comment (justinboswell) [reply CMakeLists.txt:84]: Well, these are different. One is finding a package and a named target, one is finding a library and storing it in a variable, and then the expanded path is added to the link line.

File: CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,8 @@ elseif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
     set(CMAKE_INSTALL_LIBDIR "lib")
 endif()
 
+set(FIND_LIBRARY_USE_LIB64_PATHS ON)
```
[the remark this answers is not in the document]
Comment (justinboswell) [reply CMakeLists.txt:27]: Oh good call, this is the result of some flailing. I'll pull it out and make sure we're good.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (justinboswell) [reply .github/workflows/ci.yml:84]: yes, correct. We should probably just add a `--shared` to builder that `CMakeBuild` picks up

File: CMakeLists.txt (hunk 1 above)
Comment (graebm) [reply CMakeLists.txt:82]: Sorry, I was not clear. I meant: is it normal for a find_library() call to have the side-effect of setting a variable `${AWS_LC}` that you then pass to PLATFORM_LIBS? Or is it more normal to just have the user type out both strings
```
find_library(AWS_LC crypto REQUIRED)
set(PLATFORM_LIBS dl LibCrypto::Crypto) # or whatever ${AWS_LC} expands to
```

**Review by graebm — APPROVED** [review]
fix and ship

**Pull request merged**